app/helpers/application_helper.rb

```diff
@@ -134,35 +134,39 @@ def icon_tag(type, options = {})
     content_tag :i, '', options.merge(class: css_classes)
   end
 
-  def link_to_delete(path, options = {})
-    text = options[:text] || 'Delete'
-    disabled_reason = options[:disabled]
-    if disabled_reason
-      content_tag :span, text, title: disabled_reason, class: 'mouseover'
+  def link_to_delete(
+    path, text: 'Delete', question: nil, type_to_delete: false, remove_container: false, disabled: false, **options
+  )
+    if disabled
+      content_tag :span, text, title: disabled, class: 'mouseover'
     else
-      resource = Array(path).last
       message =
-        if question = options.delete(:question)
+        if question
           question
-        elsif resource.is_a?(ActiveRecord::Base)
+        elsif path.is_a?(ActiveRecord::Base)
+          resource = path
+          name, path = link_parts_for_resource(resource)
+          "Delete #{resource.class.name.split("::").last} #{name} ?"
+        elsif (resource = Array(path).last) && resource.is_a?(ActiveRecord::Base)
```

Comment:
In this `elsif`, resource is unknown here? where is resource defined?

Reply:
resource comes from `Array(path).last)` so if path is an array then it's the last element and otherwise it's the `path`

Reply:
using `(a = 1) && 2` pattern so `a != 2`